From ba69dce89d41b0e8c38dbfd70ee34b66443635e5 Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime@cerno.tech>
Date: Fri, 15 Apr 2022 15:26:07 +0200
Subject: [PATCH] rpivid: Use clk_get_max_rate()

The driver was using clk_round_rate() to figure out the maximum clock
rate that was allowed for the HEVC clock.

Since we have a function to return it directly now, let's use it.

Signed-off-by: Maxime Ripard <maxime@cerno.tech>
---
 drivers/staging/media/rpivid/rpivid_video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/media/rpivid/rpivid_video.c b/drivers/staging/media/rpivid/rpivid_video.c
index 8007c4dbcf89..bab3ed3ff361 100644
--- a/drivers/staging/media/rpivid/rpivid_video.c
+++ b/drivers/staging/media/rpivid/rpivid_video.c
@@ -558,7 +558,7 @@ static int start_clock(struct rpivid_dev *dev, struct rpivid_ctx *ctx)
 	long max_hevc_clock;
 	int rv;
 
-	max_hevc_clock = clk_round_rate(dev->clock, ULONG_MAX);
+	max_hevc_clock = clk_get_max_rate(dev->clock);
 
 	rv = clk_set_min_rate(dev->clock, max_hevc_clock);
 	if (rv) {
-- 
2.32.0

